### ISSUE TITLE
Photos view layout mode

### ISSUE THREAD

**Issue by winston-de** [files-community/Files#5058]
**Is your feature request related to a problem?**
Feature request.

**Describe the solution**
A photos layout mode (implemented as a new data template in the grid view browser) which shows images at their original size.

**Describe alternatives you've considered**
Use grid view - large.

**Scope**

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow users to view photos in their original size | Must |
| Use adaptive layout mode feature to switch to photos view when majority or all items enumerated are images | Should |
| Show name when selected or on hover | could | 
| Sort by date taken (depends on #4432) by default when this layout mode is selected and the sort mode has not been set | Could |
| Hide preview section of preview pane and just show details since the full content is already being shown | Could |
| Add a slider to reduce image sizes | Could |

**Additional comment**
Concept from Files server:
![image](https://user-images.githubusercontent.com/59544401/120082547-77982400-c078-11eb-9b17-cc4268ff191b.png)

**Comment by yaichenbaum** [comment]
@winston-de This seems a little beyond the scope of Files. Is there added value in having this in Files over the dedicated photo app in Windows 10?

**Comment by Gmemster** [comment]
This is a must-add feature

**Comment by winston-de** [comment]
> @winston-de This seems a little beyond the scope of Files. Is there added value in having this in Files over the dedicated photos app in Windows 10?

@yair100 Although photos provides a good way to view all photos, it does not provide a good way to browse photos in specific folders, especially when those folders are not in either the pictures library or OneDrive. For example, viewing photos in a temporary downloads folder that the user may not want to keep. A pictures layout mode would make it easy to view the contents without having to import them into photos.

**Comment by mdtauk** [comment]
There was a lot of complaint and comment when the same kind of mode was removed from Windows XP's File Explorer during the Vista transition.

![winxpanim.gif](https://user-images.githubusercontent.com/7389110/120094637-6ce19b80-c119-11eb-936f-e7d1526d1ca3.gif)

**Comment by nanna66** [comment]
nice.

**Comment by ColbertBrave** [comment]
Very nice concept.

**Comment by Colbert-Ran** [comment]
I think the UI in the concept is better than the UI in the current version. Please consider the design in the concept drawing, thanks. @yair100

**Comment by nightkall** [comment]
Beautiful concept. I would love to see this mode implemented. This mode + QuickLook is all you need to explore media files.

**Comment by Rasmagoric** [comment]
> 
> 
> There was a lot of complaint and comment when the same kind of mode was removed from Windows XP's File Explorer during the Vista transition.

Interesting. I barely used windows XP and I didn't know that, this is basically the same as the finder's gallery view in mac.

But one of the reason why I designed this concept was because windows 7 had an option to hide pictures filenames.
When you have thousands of pictures with random auto-generated names, the filenames become visual clutter.

Maybe a photos view layout mode could also have more functionality with other items that are not pictures since this mode would remove the necessity to have a preview pane. (perhaps a better way to preview what's inside folders or text files... idk idk)

**Comment by R3voA3** [comment]
> @winston-de This seems a little beyond the scope of Files. Is there added value in having this in Files over the dedicated photos app in Windows 10?

I don't see how this is out of scope. Having one app to view files, folders and images it a great feature, especially since the photos app by MS is just crap.

**Comment by lukeblevins** [comment]
@yair100 I certainly think this is an area we can add value in since users want to see multiple photos at once.

**Comment by yaichenbaum** [comment]
> > @winston-de This seems a little beyond the scope of Files. Is there added value in having this in Files over the dedicated photos app in Windows 10?
> 
> I don't see how this is out of scope. Having one app to view files, folders and images it a great feature, especially since the photos app by MS is just crap.

@R3voA3 which part of this layout are you interested in, hiding the file names or is there something else?

**Comment by yair100** [comment]
> @yair100 I certainly think this is an area we can add value in since users want to see multiple photos at once.

@duke7553 How would this be different from the regular grid layout?

**Comment by R3voA3** [comment]
> > > @winston-de This seems a little beyond the scope of Files. Is there added value in having this in Files over the dedicated photos app in Windows 10?
> > 
> > 
> > I don't see how this is out of scope. Having one app to view files, folders and images it a great feature, especially since the photos app by MS is just crap.
> 
> @R3voA3 which part of this layout are you interested in, hiding the file names or is there something else?

I think it's a beautiful way to look at images without any distraction. Also, since the file name and for example sync status isn't shown more images fit on the screen at once.

**Comment by Rasmagoric** [comment]
I will give another example, when photographers are viewing photos from their folders, the pictures always have those auto generated names such as "DSC_0832",  "DSC_0833" etc, when you are trying to pick the best ones from hundreds or thousands of pictures the focus has to be on the thumbnails.

Anyways, features such as hiding the filenames or adding a scale slider could be added to the regular grid layout, it's not necessary to add another layout mode. 
But i still think that a pictures layout mode has more potential to see each item without the need for a preview pane and to see media files or any file better, or to preview what's inside folders with more detail. I think this idea should be on the table for the future.

**Comment by yaichenbaum** [comment]
> I will give another example, when photographers are viewing photos from their folders, the pictures always have those auto generated names such as "DSC_0832", "DSC_0833" etc, when you are trying to pick the best ones from hundreds or thousands of pictures the focus has to be on the thumbnails.
> 
> Anyways, features such as hiding the filenames or adding a scale slider could be added to the regular grid layout, it's not necessary to add another layout mode.
> But i still think that a pictures layout mode has more potential to see each item without the need for a preview pane and to see media files or any file better, or to preview what's inside folders with more detail. I think this idea should be on the table for the future.

I'm thinking more along the lines of having these as individual features rather than a dedicated layout mode.

**Comment by duke7553** [comment]
We don't necessarily need a new layout mode for a photos view experience, but maybe we can detect where users have all photos in a directory and tailor the existing GridViewBrowser mode to it.

**Comment by generalguy41** [comment]
Personally, I don't see the harm in adding this as another mode. If the issue is it being a dedicated option in the layouts, maybe it could be in the context menu like when you used to be able to "Hide Filenames" back in Win 7, or maybe a toggle in the settings to always show photo focused folders in this view or the original view.

**Comment by AnuthaDev** [comment]
This layout seems very similar to that of Google photos. They have published a blog describing how it works:

https://medium.com/google-design/google-photos-45b714dfbed1

Another related blog:

https://medium.com/@danrschlosser/building-the-image-grid-from-google-photos-6a09e193c74a

**Comment by AcidRain9** [comment]
This looks beautiful, is it still being considered?

**Comment by AcidRain9** [comment]
What about making it specific just to the Pictures directory of Windows?

**Comment by BanCrash** [comment]
Suggestion:

Add an option on design settings to "shows images at their original size" when detects a folder that only have images. Also, this should respect the  "enable individual preferences for folders" option, so if you have that option enabled you can enable or disable this setting for individual folders instead globally.

I mean, here:
![image](https://user-images.githubusercontent.com/11770760/141772444-de7822ac-8a65-4f48-82e8-6ddceb2da1e4.png)

**Comment by yaichenbaum** [comment]
> This proposal will allow users to view photos in their original size

@winston-de do you mean the aspect ratio?

> Show name when selected or on hover

Would this be in a tooltip or a textblock underneath the photo?

**Comment by generalguy41** [comment]
> Would this be in a tooltip or a textblock underneath the photo?

Shown over the image itself like the original concept would look nice, imo. Wouldn't have to worry about things being shifted when hovering, and could also display other things as well, like a checkbox.

That being said, since the name is already displayed in the tooltip, maybe it doesn't make sense to duplicate it.

**Comment by yaichenbaum** [comment]
> Hide preview section of preview pane and just show details since the full content is already being shown

@winston-de I think that it's still worth it to allow the preview pane in this layout since it also contains the details and meta data.
